packages/vitest/src/node/cli/cli-api.ts

```diff
@@ -184,6 +189,48 @@ export function processCollected(ctx: Vitest, files: File[], options: CliOptions
   return formatCollectedAsString(files).forEach(test => console.log(test))
 }
 
+export function outputFileList(files: WorkspaceSpec[], options: CliOptions) {
+  if (typeof options.json !== 'undefined') {
+    return outputJsonFileList(files, options)
+  }
+
+  return formatFilesAsString(files).map(file => console.log(file))
+}
+
+function outputJsonFileList(files: WorkspaceSpec[], options: CliOptions) {
+  if (typeof options.json === 'boolean') {
+    return console.log(JSON.stringify(formatFilesAsJSON(files), null, 2))
+  }
+  if (typeof options.json === 'string') {
+    const jsonPath = resolve(options.root || process.cwd(), options.json)
+    mkdirSync(dirname(jsonPath), { recursive: true })
+    writeFileSync(jsonPath, JSON.stringify(formatFilesAsJSON(files), null, 2))
+  }
+}
+
+function formatFilesAsJSON(files: WorkspaceSpec[]) {
+  return files.map((file) => {
+    const result: any = {
+      file: file.moduleId,
+    }
+
+    if (file.project.name) {
+      result.projectName = file.project.name
+    }
+    return result
+  })
+}
+
+function formatFilesAsString(files: WorkspaceSpec[]) {
+  return files.map((file) => {
+    let name = file.moduleId.substring(String(file[0].runner.root).length + 1)
```

Comment:
```suggestion
    let name = resolve(options.root || process.cwd(), file.moduleId)
```

Reply:
> 

Seems this didn't work, it prints the whole path now.